embark.el

```diff
@@ -1238,19 +1238,16 @@ Returns the name of the command."
 (defun embark--all-bindings (keymap)
   "Return an alist of all bindings in KEYMAP."
   (let (bindings)
-    (cl-labels ((gather (keymap)
-                   (map-keymap
-                    (lambda (key def)
-                      (cond
-                       ((null def))
-                       ((keymapp def)
-                        (dolist (bind (embark--all-bindings def))
-                          (push (cons (vconcat (vector key) (car bind))
-                                      (cdr bind))
-                                bindings)))
-                       (t (push (cons (vector key) def) bindings))))
-                    keymap)))
-      (gather (keymap-canonicalize keymap)))
+    (map-keymap
```

Comment:
Haha! Thanks. My code is a little ridiculous. It's an editing mistake. I was considering both your approach, and a separate less consy approach that arrange for everything to be pushed onto a single list variable (here each time you run `embark--all-bindings` you get a fresh variable `bindings`, of course). Somehow I wound up with an unholy mix of both.

Reply:
Yes, you can pull the simplification from #218, or you rewrite everything with cl-labels. I am not a cl fan :shrug:

Reply:
Merged #218 (I hadn't noticed you also had a separate PR with just that). I am a fan of cl-lib, at least of cl-loop (which is often shorter and more readable than a sequence of mapcars, seq-filters, se-find, etc.) and, occasionally, cl-labels. :)

Reply:
cl-loop is in particular what I find probematic. I rather use the composable building blocks, maps, filters, etc. However there is a point in cl-loops since if one overuses the maps and filters since one ends up with very badly performing code. But then, a few times I rewrote some cl-loops in plain imperative style with while/dolist/if/let etc and the result was not significantly longer and more readable ("python style") for beginners who are not familar with cl-loop. There is also a minor syntactic argument, maybe I would be more happy with something like [iterate](https://common-lisp.net/project/iterate/).